gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java

```diff
@@ -148,31 +147,28 @@ public void onBottomSheetHidden() {
     addObservationButtonVisibility.setValue(View.GONE);
   }
 
-  public void onFormChange(Form form) {
-    this.selectedForm = form;
-  }
-
   public void addObservation() {
     FeatureSheetState state = featureSheetState.getValue();
     if (state == null) {
       Log.e(TAG, "Missing featureSheetState");
       return;
     }
-    if (selectedForm == null) {
-      Log.e(TAG, "Missing form");
-      return;
-    }
     Feature feature = state.getFeature();
     if (feature == null) {
       Log.e(TAG, "Missing feature");
       return;
     }
+    ImmutableList<Form> forms = feature.getLayer().getForms();
+    if (forms.isEmpty()) {
```

Comment:
Add TODO to show a message to the user?

Reply:
Added TODO to hide Add Observation button instead, since it shouldn't be shown if form is not defined (i.e., this case should never happen).